### PR TITLE
CUDA BFloat16 batchnorm (non-cuDNN)

### PR DESCRIPTION
**Pull request by zasdfgbnm**

### CODE REVIEW

**Review by ngimel — APPROVED** [review]

**Comment by ngimel** [comment]
Can you please rebase to get CI signal? Otherwise looks good. We have so annoyingly many helper functions for batch norm.

**Comment by zasdfgbnm** [comment]
@ngimel Rebased
